backend/canisters/identity/impl/src/model/identity_map.rs

```diff
@@ -1,15 +1,17 @@
 use candid::Principal;
-// use phonenumber::PhoneNumber;
-// use std::collections::hash_map;
-// use std::collections::hash_map::Entry::Vacant;
+use std::collections::hash_map::Entry::{Occupied, Vacant};
 use std::collections::HashMap;
+use std::collections::HashSet;
 use types::Identity;
-// use types::TimestampMillis;
+use types::TimestampMillis;
+use types::VerifiableField;
+use types::VerificationCodeStatus;
+use types::{FieldId, PhoneNumber};
 
 #[derive(Default)]
 pub struct IdentityMap {
     identities_by_principal: HashMap<Principal, Identity>,
-    // phone_number_to_principal: HashMap<PhoneNumber, Principal>,
+    registered_phone_numbers: HashSet<PhoneNumber>,
```

Comment:
We need a way of allowing phone numbers to be recycled. We could store the date each number is registered and allow them to be reused after 3 months or so.

Reply:
Not a priority for the hackathon I suppose...